src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ListItemViewModel.cs

```diff
@@ -14,22 +14,70 @@ public partial class ListItemViewModel(IListItem model, TaskScheduler scheduler)
 
     // Remember - "observable" properties from the model (via PropChanged)
     // cannot be marked [ObservableProperty]
-    public ITag[] Tags { get; private set; } = [];
+    public List<TagViewModel> Tags { get; private set; } = [];
 
-    public bool HasTags => Tags.Length > 0;
+    public bool HasTags => Tags.Count > 0;
+
+    public string TextToSuggest { get; private set; } = string.Empty;
+
+    public string Section { get; private set; } = string.Empty;
 
     public override void InitializeProperties()
     {
         base.InitializeProperties();
 
-        // TODO load tags here, details, suggested text, all that
         var li = _listItemModel.Unsafe;
         if (li == null)
         {
             return; // throw?
         }
 
-        // TODO TagViewModel not ITag
-        Tags = li.Tags ?? [];
+        Tags = li.Tags?.Select(t =>
+        {
+            var vm = new TagViewModel(t, Scheduler);
+            vm.InitializeProperties();
+            return vm;
+        })
+            .ToList() ?? [];
+        TextToSuggest = li.TextToSuggest;
+        Section = li.Section ?? string.Empty;
+
+        UpdateProperty(nameof(HasTags));
+        UpdateProperty(nameof(Tags));
+        UpdateProperty(nameof(TextToSuggest));
+        UpdateProperty(nameof(Section));
+    }
+
+    protected override void FetchProperty(string propertyName)
```

Comment:
We're really only having to do this because of all our marshalling requirements, eh? This does seem like a case since we're doing this all the time where a source generator would just make sense to do this type of work for us for the base cases, but that's a lot of effort to prop up... just a future note...

Reply:
I've had the same thought. There's gonna be a lot more of these - I'm guessing we'll write the source generator inevitably